plugin/rfactory.vim

```diff
@@ -1,30 +1,89 @@
-if !exists("g:rfactory_factory_location")
-  let g:rfactory_factory_location = "spec/factories.rb"
-endif
-
 let rfactory_fg_methods = ['create', 'build', 'build_stubbed', 'attributes_for']
 let method_pattern = '\<\%('. join(rfactory_fg_methods, '\|') . '\)' " Non capturing 'or'
 let symbol_pattern = '\(:\w\+\)'
 let optional_trait_pattern = '\%(, '.symbol_pattern.'\)\?'
 let s:pattern = method_pattern . '.' . symbol_pattern . optional_trait_pattern
+let s:FALSE = 0
+let s:TRUE = 1
+let s:factory_path_patterns = [
+      \ "spec/factories.rb",
+      \ "test/factories.rb",
+      \ "spec/factories/*.rb",
+      \ "test/factories/*.rb"
+      \ ]
 
 function! s:FactoryOnCurrentLine()
   return matchlist(getline('.'), s:pattern)
 endfunction
 
-function! s:Rfactory(edit_method)
-  let factory = s:FactoryOnCurrentLine()
-  if len(factory)
-    let factory_name = factory[1]
-    let factory_trait = factory[2]
-    execute a:edit_method . ' ' . g:rfactory_factory_location
-    call search('.*factory.' . factory_name)
-    if factory_trait !=? ''
-      call search('.*trait.' . factory_trait)
+function! s:FactoryFilePaths()
+  let results = []
+  for pattern in s:factory_path_patterns
+    call extend(results, s:FilesForGlobAsList(pattern))
+  endfor
+  return results
+endfunction
+
+function! s:FilesForGlobAsList(glob_pattern)
+  return glob(a:glob_pattern, 0, 1)
+endfunction
+
+function! s:BuildFactoryContext()
+  let factory_definition = s:FactoryOnCurrentLine()
+  let factory = { "is_present": len(factory_definition) }
+  if factory.is_present
+    let factory["name"] = factory_definition[1]
+    let factory["trait"] = factory_definition[2]
+  endif
+  return factory
+endfunction
+
+function! s:TryToFindTrait(trait)
+  if a:trait != ''
+    call search('.*trait.' . a:trait)
+  endif
+endfunction
+
+function! s:CenterFactoryInWindow()
+  normal! zz
+endfunction
+
+function! s:FactoryDefinitionFound(factory_name)
+  return search('.*factory.' . a:factory_name) != 0
```

Comment:
Looks like double quotes and single quotes are used interchangeably in this file. Does Vim care?